Bottom: b35c5345d988f433eec8fb8e50d196d6f0f7c9c2
Top:    22eb8feda6e89fa38358a6b7b875d27af8a4f5e0
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2008-09-29 22:23:57 +0100

Refresh of drop-in-pcb-gl-code-various-me

---

diff --git a/src/hid/gtk/gtkhid-main.c b/src/hid/gtk/gtkhid-main.c
index bf9957c..6accef8 100644
--- a/src/hid/gtk/gtkhid-main.c
+++ b/src/hid/gtk/gtkhid-main.c
@@ -938,6 +938,7 @@ ghid_draw_line (hidGC gc, int x1, int y1, int x2, int y2)
 {
 #define TRIANGLES_PER_CAP 15
 #define MIN_TRIANGLES_PER_CAP 3
+#define MAX_TRIANGLES_PER_CAP 1000
   double dx1, dy1, dx2, dy2;
   double width, angle;
   float deltax, deltay, length;
@@ -991,6 +992,9 @@ ghid_draw_line (hidGC gc, int x1, int y1, int x2, int y2)
   if (slices < MIN_TRIANGLES_PER_CAP)
     slices = MIN_TRIANGLES_PER_CAP;
 
+  if (slices > MAX_TRIANGLES_PER_CAP)
+    slices = MAX_TRIANGLES_PER_CAP;
+
 //  slices = TRIANGLES_PER_CAP;
 
   switch (gc->cap) {
@@ -1165,6 +1169,7 @@ ghid_fill_circle (hidGC gc, int cx, int cy, int radius)
 {
 #define TRIANGLES_PER_CIRCLE 30
 #define MIN_TRIANGLES_PER_CIRCLE 6
+#define MAX_TRIANGLES_PER_CIRCLE 2000
   gint w, h, vx, vy, vr;
   float last_x, last_y;
   int slices;
@@ -1188,6 +1193,9 @@ ghid_fill_circle (hidGC gc, int cx, int cy, int radius)
   if (slices < MIN_TRIANGLES_PER_CIRCLE)
     slices = MIN_TRIANGLES_PER_CIRCLE;
 
+  if (slices > MAX_TRIANGLES_PER_CIRCLE)
+    slices = MAX_TRIANGLES_PER_CIRCLE;
+
 //  slices = TRIANGLES_PER_CIRCLE;
 
   ensure_triangle_space (slices);
